tests/rheology_scaled_profile.cc

```diff
@@ -0,0 +1,275 @@
+     * are defined in dependence of position.
+     */
+    template<int dim>
+    class ScaledViscosityProfileMaterial : public MaterialModel::Interface<dim>, public aspect::SimulatorAccess<dim>
+    {
+      public:
+        void update() override
+        {
+          reference_viscosity_coordinates = reference_viscosity_profile->get_coordinates();
+
+          std::vector<std::unique_ptr<internal::FunctorBase<dim> > > lateral_averaging_properties;
+          lateral_averaging_properties.push_back(std::make_unique<internal::FunctorDepthAverageUnscaledViscosity<dim>>());
+
+          std::vector<std::vector<double>> averages =
+                                          this->get_lateral_averaging().compute_lateral_averages(reference_viscosity_coordinates,
+                                              lateral_averaging_properties);
+
+          laterally_averaged_viscosity_profile.swap(averages[0]);
+
+          for (unsigned int i = 0; i < laterally_averaged_viscosity_profile.size(); ++i)
+            AssertThrow(numbers::is_finite(laterally_averaged_viscosity_profile[i]),
+                        ExcMessage("In computing depth averages, there is at"
+                                   " least one depth band that does not have"
+                                   " any quadrature points in it."
+                                   " Consider reducing number of depth layers"
+                                   " for averaging."));
+        }
+
+
+
+        double
+        compute_viscosity_scaling (const double depth) const
+        {
+          // Make maximal depth slightly larger to ensure depth < maximal_depth
+          const double maximal_depth = this->get_geometry_model().maximal_depth() *
+                                       (1.0+std::numeric_limits<double>::epsilon());
+          Assert(depth < maximal_depth, ExcInternalError());
+
+          unsigned int depth_index;
+          if (depth < reference_viscosity_coordinates.front())
+            {
+              depth_index = 0;
+            }
+          else if (depth > reference_viscosity_coordinates.back())
+            {
+              depth_index = reference_viscosity_coordinates.size() - 1;
+            }
+          else
+            {
+              depth_index = std::distance(reference_viscosity_coordinates.begin(),
+                                          std::lower_bound(reference_viscosity_coordinates.begin(),reference_viscosity_coordinates.end(),depth));
+              if (depth_index > 0)
+                depth_index -= 1;
+            }
+
+          double reference_viscosity;
+
+          if (false)
+            {
+              reference_viscosity = reference_viscosity_profile->compute_viscosity(depth);
+            }
+          else
+            {
+              reference_viscosity = reference_viscosity_profile->compute_viscosity(reference_viscosity_coordinates.at(depth_index));
```

Comment:
I am a bit confused how this works. So the ascii data file in the AsciiDepthProfile rheology has to contain a number of coordinates, and a number of viscosities, and I guess they have to have the same number of entries? So then is the viscosity the one above or below the coordinate? This should be documented somewhere.